diff --git a/src/RosCard.js b/src/RosCard.js
index 548c7ab..3e1f47b 100644
--- a/src/RosCard.js
+++ b/src/RosCard.js
@@ -591,7 +591,7 @@
                                             i = r[0],
                                             a = r[1],
                                             c = document.createElement("div");
-                                        c.innerHTML = "<span>".concat(t.entity_id, ": ").concat(i, "</span>");
+                                        c.appendChild(Object.assign(document.createElement("span"), { textContent: "".concat(t.entity_id, ": ").concat(i) }));
                                         var u = e._createButton(e._translations[e._language].sendButton, function() {
                                             "volume_mute" === a ? e._hass.callService(t.type, a, {
                                                 entity_id: t.entity_id,
@@ -606,7 +606,7 @@
                                             i = r[0],
                                             a = r[1],
                                             c = document.createElement("div");
-                                        c.innerHTML = "<span>".concat(t.entity_id, ": ").concat(i, " = ").concat(a, "</span>");
+                                        c.appendChild(Object.assign(document.createElement("span"), { textContent: "".concat(t.entity_id, ": ").concat(i, " = ").concat(a) }));
                                         var u = e._createButton(e._translations[e._language].testButton, function() {
                                             e._hass.callService(t.type, "send_command", {
                                                 entity_id: t.entity_id,
@@ -619,7 +619,7 @@
                                             i = r[0],
                                             a = r[1],
                                             c = document.createElement("div");
-                                        c.innerHTML = "<span>".concat(t.entity_id, ": ").concat(i, " = ").concat(a, "</span>");
+                                        c.appendChild(Object.assign(document.createElement("span"), { textContent: "".concat(t.entity_id, ": ").concat(i, " = ").concat(a) }));
                                         var u = e._createButton(e._translations[e._language].testButton, function() {
                                             e._hass.callService(t.type, "select_option", {
                                                 entity_id: t.entity_id,
@@ -6908,7 +6908,7 @@
                                 n.style.padding = "16px", n.style.position = "relative", this._config.entities.length > 0 ? this._config.entities.forEach(function(t) {
                                     var r = e._getEntityDisplayName(t),
                                         o = document.createElement("div");
-                                    o.style.marginBottom = "10px", o.innerHTML = "<div>".concat(e._translations[e._language].entity, ": ").concat(r, "</div>"), n.appendChild(o)
+                                    o.style.marginBottom = "10px", o.appendChild(Object.assign(document.createElement("div"), { textContent: "".concat(e._translations[e._language].entity, ": ").concat(r) })), n.appendChild(o)
                                 }) : n.innerHTML = "<div>".concat(this._translations[this._language].validEntity, "</div>"), t.appendChild(n), this.appendChild(t)
                             }
                         }
