diff --git a/src/RosCard.js b/src/RosCard.js
index 82da619..4e1256b 100644
--- a/src/RosCard.js
+++ b/src/RosCard.js
@@ -562,7 +562,7 @@
                             this._config = d(d({}, e), {}, {
                                 tv_name: e.tv_name || ("zh" === this._language ? "未命名电视" : "Unnamed TV"),
                                 entities: Array.isArray(null == e ? void 0 : e.entities) ? u(e.entities) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_tv.png",
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_tv.png",
                                 tv_type: e.tv_type || "android_tv"
                             }), this._hass && this.render()
                         }
@@ -2135,7 +2135,7 @@
                         value: function(e) {
                             this._config = re(re({}, e), {}, {
                                 entities: Array.isArray(null == e ? void 0 : e.entities) ? ee(e.entities) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_light.png"
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_light.png"
                             }), this._hass && this.render()
                         }
                     }, {
@@ -3192,7 +3192,7 @@
                         value: function(e) {
                             this._config = ze(ze({}, e), {}, {
                                 entities: Array.isArray(null == e ? void 0 : e.entities) ? He(e.entities) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_fan.png"
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_fan.png"
                             }), this._hass && this.render()
                         }
                     }, {
@@ -3624,7 +3624,7 @@
                                         image_name: e.image_name || ""
                                     })
                                 }) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_scene.png"
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_scene.png"
                             }), this._hass && this.render()
                         }
                     }, {
@@ -4564,7 +4564,7 @@
                         value: function(e) {
                             this._config = Mt(Mt({}, e), {}, {
                                 entities: Array.isArray(null == e ? void 0 : e.entities) ? Bt(e.entities) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_media_play1.png"
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_media_play1.png"
                             }), this._hass && this.render()
                         }
                     }, {
@@ -5059,7 +5059,7 @@
                         value: function(e) {
                             this._config = cn(cn({}, e), {}, {
                                 entities: Array.isArray(null == e ? void 0 : e.entities) ? rn(e.entities) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_climate.png"
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_climate.png"
                             }), this._hass && this.render()
                         }
                     }, {
@@ -5499,7 +5499,7 @@
                         value: function(e) {
                             this._config = Tn(Tn({}, e), {}, {
                                 entities: Array.isArray(null == e ? void 0 : e.entities) ? Cn(e.entities) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_cover.png"
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_cover.png"
                             }), this._hass && this.render()
                         }
                     }, {
@@ -5944,7 +5944,7 @@
                         value: function(e) {
                             this._config = Xn(Xn({}, e), {}, {
                                 entities: Array.isArray(null == e ? void 0 : e.entities) ? qn(e.entities) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_switch.png"
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_switch.png"
                             }), this._hass && this.render()
                         }
                     }, {
@@ -6377,7 +6377,7 @@
                         value: function(e) {
                             this._config = gr(gr({}, e), {}, {
                                 entities: Array.isArray(null == e ? void 0 : e.entities) ? vr(e.entities) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_weather.png"
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_weather.png"
                             }), this._hass && this.render()
                         }
                     }, {
@@ -6888,7 +6888,7 @@
                         value: function(e) {
                             this._config = Ur(Ur({}, e), {}, {
                                 entities: Array.isArray(null == e ? void 0 : e.entities) ? Mr(e.entities) : [],
-                                icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_host.png"
+                                icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_host.png"
                             }), this._hass && this.render()
                         }
                     }, {
@@ -7291,7 +7291,7 @@
                             value: function(e) {
                                 this._config = co(co({}, e), {}, {
                                     id: e.id || this.generateUUID(),
-                                    icon_path: e.icon_path || "/local/community/RosCard/icon_img/icon_statistics.png",
+                                    icon_path: e.icon_path || "/local/community/roscard_hardened/icon_img/icon_statistics.png",
                                     monitor_name: e.monitor_name || ("zh" === this._language ? "未命名监听" : "Unnamed Monitor"),
                                     device_types: co({
                                         light: !0,
@@ -7512,7 +7512,7 @@
                         validEntity: "请选择有效实体",
                         stateOn: "开",
                         stateOff: "关",
-                        defaultIconPath: "/local/community/RosCard/icon_img/states_device_light_on_icon.png",
+                        defaultIconPath: "/local/community/roscard_hardened/icon_img/icon_light.png",
                         entity: "实体",
                         executionModes: {
                             immediate: "立即执行",
@@ -7616,7 +7616,7 @@
                         validEntity: "Please select a valid entity",
                         stateOn: "On",
                         stateOff: "Off",
-                        defaultIconPath: "/local/community/RosCard/icon_img/states_device_light_on_icon.png",
+                        defaultIconPath: "/local/community/roscard_hardened/icon_img/icon_light.png",
                         entity: "Entity",
                         executionModes: {
                             immediate: "Immediate",
@@ -7849,7 +7849,9 @@
                         key: "_createIcon",
                         value: function(e) {
                             var t = document.createElement("img");
-                            return t.src = "string" == typeof e && /^\/[^\/\\]/.test(e) ? e : this._translations[this._language].defaultIconPath, t.style.position = "absolute", t.style.top = "12px", t.style.right = "16px", t.style.width = "60px", t.style.height = "60px", t.style.opacity = "0.8", t.style.pointerEvents = "none", t
+                            return t.onerror = function() {
+                                t.style.display = "none"
+                            }, t.src = "string" == typeof e && /^\/[^\/\\]/.test(e) ? e : this._translations[this._language].defaultIconPath, t.style.position = "absolute", t.style.top = "12px", t.style.right = "16px", t.style.width = "60px", t.style.height = "60px", t.style.opacity = "0.8", t.style.pointerEvents = "none", t
                         }
                     }, {
                         key: "_t",
